### ISSUE TITLE
Element type and status should be pronounced in voice language, not in NVDA language

### ISSUE THREAD

**Issue by Adriani90** [nvaccess/nvda#9150]
### Steps to reproduce:
1. Open this website where you have three languages (german, english and some french)
http://codepen.io/jmuheim/pen/doVPqg
Note: this issue occurs also in MS Excel and potentially other programs with slightly different stepts to reproduce.
2. Open NVDA with NVDA language set on english
3. Enable automatic switch language and automatic recognize dialects in synth settings
3. Enable table and column / ro headings in document formating settings
4. Switch the voice language to i.e. finish or spanish

### Actual behavior:
NVDA recognizes automatically german, english and french language and switches the voice accordingly for text and element labels. But element type such as button, edit field etc is read in NVDA language instead of voice language. So, if voice language is spanish and NVDA language is english, the text on this website will be read according to its language but the element type will be read in english although voice language is spanish.

### Expected behavior:
Element types and status should be reported in voice language, not in NVDA language.

#### NVDA installed/portable/running from source:
Installed

#### NVDA version:
2018.4

#### Windows version:
Windows 10 1809 Update

#### Name and version of other software in use when reproducing the issue:
Firefox 64.0.2

#### Does the issue still occur after restarting your PC?
yes

#### Have you tried any other versions of NVDA?
yes, 2017.3

**Comment by Adriani90** [comment]
an argument for solving this issue is that often people do not use automatic language detection and prefer to change the voice language to read parts of a text which is in a different language. But if you don't enable automatically switch language, then the voice language will pronounce the types and status of elements. The rponounciation is then broken because the voice language is not the same like the nvda language.

**Comment by leonardder** [comment]
I have no idea whether gettext is able to do this. cc @josephsl

**Comment by ruifontes** [comment]
Hello!

Just a question:
If the NVDA language is set to english, NVDA will be using the NVDA.MO 
present in the EN folder of Locale folder, so all elements and status 
are in english...

To have NVDA using the elements and status in the voice language, it 
will be necessary to allow NVDA to read the NVDA.MO from the Locale 
folder corresponding to the voice used...

Rui Fontes



Às 18:03 de 13/01/2019, Adriani90 escreveu:

**Comment by lukaszgo1** [comment]
It works exactly the same way in *all* other screen readers. The current behavior is the expected one. If your screen reader has language set to English you want to hear the control types inn the language in which my screen reader is.

**Comment by Adriani90** [comment]
Yes this is expected as long as you enable automatic language detection. But if not, the pronounciation is broken. I assume users who use a different voice language they can understand it. It‘s more confortable to have a convenient pronounciation. Especially when the voice rate is quite high.

Von meinem iPhone gesendet

**Comment by lukaszgo1** [comment]
The pronunciation is broken, because user explicitly choose to broke it. It should remain that way. We really donot want to replicate the poor experience which VoiceOver offers on iOS when voice language is changed in the rotor.

**Comment by josephsl** [comment]
Hi,

One way to partially resolve this is editing the user guide to point out the fact that the option mentioned deals with document content, not necessarily element types and states.

With that said, object roles and states and their translations come from NVDA, not speech synthesizers (I know this because I regularly deal with English and Korean and switch to Nuance Vocalizer for reading Korean, and currently I told NVDA to not switch languages automatically). Yes, it is helpful to tell NVDA to switch languages automatically when encountering text written and marked up in the target language correctly (mostly falling under web authoring issue), but I'd say it might not be a good long-term idea to enforce this for roles and states also. Also, not all speech synthesizers come with voices for some languages NVDA has been translated into, nor some of them do not expose language metadata in a way that can be useful for this situation.

Thus, just like what Lukas said, I'm in favor of closing this (as wontfix). I'll however leave open the possibility of incorporating the clarification I mentioned above as part of docs19.1 branch (once 2019.1 beta 1 ships).

Thanks.

**Comment by leonardder** [comment]
> Thus, just like what Lukas said, I'm in favor of closing this (as wontfix). I'll however leave open the possibility of incorporating the clarification I mentioned above as part of docs19.1 branch (once 2019.1 beta 1 ships).

I don't think it makes sense to have the clarification as part of the docs 19.1 pr. Let's leave that to changes that really need to be done before 2019.1.

Closing as won't fix, as I agree with the above comments.
